standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -1493,6 +1497,39 @@ public static String getPartitionName(Path tablePath, Path partitionPath, Set<St
     return result;
   }
 
+  public static String getNormalisedPartitionValue(String partitionValue, String type) {
+
+    LOG.debug("Converting '" + partitionValue + "' to type: '" + type + "'.");
+
+    if (type.equalsIgnoreCase("tinyint")
+    || type.equalsIgnoreCase("smallint")
+    || type.equalsIgnoreCase("int")){
+      return Integer.toString(Integer.parseInt(partitionValue));
+    } else if (type.equalsIgnoreCase("bigint")){
+      return Long.toString(Long.parseLong(partitionValue));
+    } else if (type.equalsIgnoreCase("float")){
+      return Float.toString(Float.parseFloat(partitionValue));
+    } else if (type.equalsIgnoreCase("double")){
+      return Double.toString(Double.parseDouble(partitionValue));
+    } else if (type.startsWith("decimal")){
+      // Decimal datatypes are stored like decimal(10,10)
+      return new BigDecimal(partitionValue).stripTrailingZeros().toPlainString();
+    }
+    return partitionValue;
+  }
+
+  public static Map<String, String> getPartitionColtoTypeMap(List<FieldSchema> partitionCols) {
+    Map<String, String> typeMap = new HashMap<>();
+
+    if (!(partitionCols == null || partitionCols.isEmpty())) {
```

Comment:
Avoid outer negation and use if ((partitionCols != null) && !partitionCols.isEmpty()). Even isEmpty need not be checked here as below loop won't execute if it is empty.